### ISSUE TITLE
Significant performance gap between the JAX and PyTorch models

### ISSUE THREAD

**Issue by JinmingM** [Physical-Intelligence/openpi#824]
Hi, I observed a significant performance gap between the JAX and PyTorch models.

When training with the same real-world dataset and the same data normalization, the JAX version achieves good performance, but the PyTorch version performs very poorly.

I followed the training pipeline described in the README (including checkpoint covert, data loading, normalization, and all default hyperparameters).

Could you please help check whether there are known inconsistencies between the JAX and PyTorch implementations, or if there are additional steps required to reproduce the JAX results using PyTorch?

Thanks!

**Comment by evelynmitchell** [comment]
Are you familiar with profiling tools such as flamegraphs?
https://www.brendangregg.com/flamegraphs.html

If you collect traces from Jax and PyTorch that would help narrow down the performance gap.

In any case, this is likely something that needs to be addressed by the Jax or PyTorch teams, not this project.

**Comment by lucasjinreal** [comment]
@JinmingM Did you mean the success rate? Can u post any further evidence indicates that pytorch version is worse with jax? AFAIK, the models weights produce same floats with same input.
